src/jdk.jfr/share/classes/jdk/jfr/internal/PlatformRecorder.java

```diff
@@ -498,17 +508,24 @@ private void periodicTask() {
             return;
         }
         while (true) {
-            synchronized (this) {
-                if (jvm.shouldRotateDisk()) {
-                    rotateDisk();
-                }
-                if (isToDisk()) {
-                    EventLog.update();
+            long wait = Options.getWaitInterval();
+            try {
+                synchronized (this) {
+                    if (jvm.shouldRotateDisk()) {
+                        rotateDisk();
+                    }
+                    if (isToDisk()) {
+                        EventLog.update();
+                    }
                 }
+                long minDelta = PeriodicEvents.doPeriodic();
+                wait = Math.min(minDelta, Options.getWaitInterval());
+            } catch(Throwable t) {
+                // Catch everything and log, but don't allow it to end the periodic task
+                Logger.log(JFR_SYSTEM, ERROR, "Error in Periodic task: " + t.getClass().getName() + ", " + t.getMessage());
```

Comment:
Not sure if it matters, but you could get an OOM when string are concatenated and the thread will end.